e2e/playwright/search/src/tests/search-filters-categories.e2e.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { expect } from '@playwright/test';
+import { ApiClientFactory, Utils, test, NodesApi, TrashcanApi, FileActionsApi, TEST_FILES, CategoriesApi } from '@alfresco/playwright-shared';
+import { CategoryLinkBody, CategoryPaging, CategoryEntry } from '@alfresco/js-api';
+
+test.describe('Search - Filters - Categories', () => {
+  let nodesApi: NodesApi;
+  let trashcanApi: TrashcanApi;
+  let categoriesApi: CategoriesApi;
+  let categoryData: CategoryPaging | CategoryEntry;
+  let categoryId: string;
+
+  const randomId = Utils.random();
+  const username = `user-${randomId}`;
+  const fileNamePdf = `${randomId}-fileNameKb.pdf`;
+  const fileNameJpg = `${randomId}-fileNameMb.jpg`;
+
+  const newSubcategories = [{ name: 'e2e' }];
+
+  test.beforeEach(async ({ loginPage }) => {
+    await Utils.tryLoginUser(loginPage, username, username, 'beforeEach failed');
+  });
+
+  test.beforeAll(async () => {
+    try {
+      const apiClientFactory = new ApiClientFactory();
+      await apiClientFactory.setUpAcaBackend('admin');
+      await apiClientFactory.createUser({ username });
+      nodesApi = await NodesApi.initialize(username, username);
+      trashcanApi = await TrashcanApi.initialize(username, username);
+      categoriesApi = await CategoriesApi.initialize('admin');
+      const fileActionsApi = await FileActionsApi.initialize(username, username);
+      await fileActionsApi.uploadFileWithRename(TEST_FILES.PDF.path, fileNamePdf, '-my-');
+      const jpgFileId = (await fileActionsApi.uploadFileWithRename(TEST_FILES.JPG_FILE.path, fileNameJpg, '-my-')).entry.id;
+      categoryData = await categoriesApi.createCategory('-root-', newSubcategories);
+      if (categoryData && 'entry' in categoryData) {
```

Comment:
What are you trying to check here? If you need to check the type you can use `instanceof` or if you want to check if `categoryData` has `entry` property you can use `categoryData?.entry`

Reply:
This is the only way I came up with that let me use ".entry.id" on "categoryData". The main issue here is that "categoryData" has the "CategoryPaging | CategoryEntry" type, and only "CategoryEntry" lets the use of ".entry.id".

I tried many different iterations to make it look nice, but this is the only one that works and keeps the "CategoryPaging | CategoryEntry" type.

I tried using "categoryData?.entry" but the "CategoryPaging" does let that.

Using "instanceof" doesnt change much so I am not sure here.

Reply:
I think we don't need this if check, createCategory return entry or it will fail. this we are already catch createCategory function.

Reply:
Using `instanceof` verifies if passed object has correct type hence has correct properties so in this case I think the correct option would be to use `categoryData != null && categoryData instanceof CategoryEntry`

Reply:
Without this if I get this error:
"Property 'entry' does not exist on type 'CategoryPaging | CategoryEntry'.
  Property 'entry' does not exist on type 'CategoryPaging'."

Reply:
If is required in this case, like I said you need to use this condition: `categoryData != null && categoryData instanceof CategoryEntry`

Reply:
I updated with `categoryData != null && categoryData instanceof CategoryEntry`.

Reply:
correction, I updated with `if (categoryData instanceof CategoryEntry) {`